Formula/flutter.rb

```diff
@@ -0,0 +1,50 @@
+class Flutter < Formula
+  desc "Build beautiful mobile apps easily and quickly"
+  homepage "https://flutter.io"
+  url "https://storage.googleapis.com/flutter_infra/releases/beta/macos/flutter_macos_v0.3.1-beta.zip"
+  sha256 "a88356eb37c37e89f92698ed06eaf02d6364c50f6accea6fe793b9d04c9e0c98"
+
+  def install
+    # Flutter needs to be in a full git clone of its repository in order to run.
+    # Thankfully, that's included in the tarball/zip, so just copy the contents:
+    cp_r ".", prefix
+  end
+
+  def post_install
+    # Something about Homebrew's post-install process removes write permissions
+    # and Flutter writes to bin/cache, particularly .lockfile, so reenable
+    # write permissions on the cache folder
+    chmod_R "u+w", prefix/"bin/cache"
+    # See https://github.com/Homebrew/homebrew-core/issues/17098#issuecomment-386031418
```

Comment:
This is a valid alternative to the double upgrading conflict. We can actually prompt the user back to use `brew upgrade` if the directory is not writable from the flutter tool. Feel free to file a feature request for us for that handling. 

We also want users to be able to modify packages/flutter/lib and modify the framework dart code.